player/cPlayerDifficultySettings.cpp

```diff
@@ -27,23 +28,23 @@ cPlayerDifficultySettings::~cPlayerDifficultySettings() {
 }
 
 float cPlayerDifficultySettings::getMoveSpeed(int iUnitType, int slowDown) {
-    return (int)(sUnitInfo[iUnitType].speed) + ((float)slowDown * m_moveSpeedFactor);
+    return sUnitInfo[iUnitType].speed + slowDown * m_moveSpeedFactor;
 }
 
 float cPlayerDifficultySettings::getBuildSpeed(int iSpeed) {
-    return (int) (iSpeed * m_buildSpeedFactor);
+    return std::floor(iSpeed * m_buildSpeedFactor);
```

Comment:
Was this the intent? If yes, this makes it more explicit. If no, we found a bug. :)